adapters/mobilefuse/mobilefuse.go

```diff
@@ -0,0 +1,193 @@
+
+	if err := json.Unmarshal(response.Body, &incomingBidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	outgoingBidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	for _, seatbid := range incomingBidResponse.SeatBid {
+		for i := range seatbid.Bid {
+			outgoingBidResponse.Bids = append(outgoingBidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatbid.Bid[i],
+				BidType: adapter.getBidType(seatbid.Bid[i].ImpID, incomingRequest.Imp),
+			})
+		}
+	}
+
+	return outgoingBidResponse, nil
+}
+
+func (adapter *MobilefuseAdapter) makeRequest(bidRequest *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	mobilefuseBidRequest := *bidRequest
+
+	mobilefuseExtension, errs := adapter.getMobilefuseExtension(bidRequest)
+
+	if mobilefuseExtension == nil {
+		return nil, errs
+	}
+
+	endpoint, err := adapter.getEndpoint(mobilefuseExtension)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	adapter.modifyBidRequest(&mobilefuseBidRequest, mobilefuseExtension)
+
+	if len(mobilefuseBidRequest.Imp) == 0 {
+		err := fmt.Errorf("No valid imps")
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	body, err := json.Marshal(mobilefuseBidRequest)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endpoint,
+		Body:    body,
+		Headers: headers,
+	}, errs
+}
+
+func (adapter *MobilefuseAdapter) getMobilefuseExtension(request *openrtb.BidRequest) (*openrtb_ext.ExtImpMobilefuse, []error) {
+	var mobilefuseImpExtension openrtb_ext.ExtImpMobilefuse
+	var errs []error
+
+	for _, imp := range request.Imp {
+		var bidder_imp_extension adapters.ExtImpBidder
+
+		err := json.Unmarshal(imp.Ext, &bidder_imp_extension)
+
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		err = json.Unmarshal(bidder_imp_extension.Bidder, &mobilefuseImpExtension)
+
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		break
+	}
+
+	return &mobilefuseImpExtension, errs
+}
+
+func (adapter *MobilefuseAdapter) getEndpoint(ext *openrtb_ext.ExtImpMobilefuse) (string, error) {
+	publisher_id := strconv.Itoa(ext.PublisherId)
+
+	url, errs := macros.ResolveMacros(adapter.EndpointTemplate, macros.EndpointTemplateParams{PublisherID: publisher_id})
+
+	if errs != nil {
+		return "", errs
+	}
+
+	if ext.TagidSrc == "ext" {
+		url += "&tagid_src=ext"
+	}
+
+	return url, nil
+}
+
+func (adapter *MobilefuseAdapter) modifyBidRequest(bidRequest *openrtb.BidRequest, ext *openrtb_ext.ExtImpMobilefuse) {
+	var validImps []openrtb.Imp
+
+	for _, imp := range bidRequest.Imp {
+		if imp.Banner != nil || imp.Video != nil {
+			if imp.Banner != nil && imp.Video != nil {
+				imp.Video = nil
+			}
+
+			imp.TagID = strconv.Itoa(ext.PlacementId)
+			imp.Ext = nil
+			validImps = append(validImps, imp)
+		}
+
+		break
```

Comment:
`break` statment in line 181 opens up the possibility of returning no `validImps` even if only the first element of the `[]Imp` slice is the only one to come with both `Banner` and `Video` set to `nil`. Is the scenario below possible in the context of your adapter?
```
bidRequest.Imp = []Imp{
    { //first element
        Banner:nil,
        Video:nil
    },
    { //second element
        Banner: /* a valid Banner*/
        Video: /* a valid Video*/
    },
}
```
Are we sure we don't want to check the rest of the []`Imp` slice for valid elements?

Reply:
Nice catch, this was a dumb mistake.

Reply:
It's not dumb. It happens